word_language_model/main.py

```diff
@@ -92,7 +109,17 @@ def batchify(data, bsz):
 ###############################################################################
 
 ntokens = len(corpus.dictionary)
-model = model.RNNModel(args.model, ntokens, args.emsize, args.nhid, args.nlayers, args.dropout, args.tied).to(device)
+if args.model == 'Transformer':
+	model = model.TransformerSeq2Seq(ntokens, ntokens, nhead=args.transformer_head, 
+                                     d_model=args.emsize, dropout=args.dropout, 
+                                     num_encoder_layers=args.transformer_encoder_layers, 
+                                     num_decoder_layers=args.transformer_decoder_layers, 
+                                     dim_feedforward=args.transformer_dim_feedforward)
+	### Change the generator in transformer to nn.Linear(d_model, vocab)
+	model.generator = nn.Linear(args.emsize, ntokens)
```

Comment:
Move this into the initialization. Replacing member variables after instantiation can cause pain further down the road, because you're not in control of your state.

Reply:
You initialize model.generator with an instance of the Generator class, but then overwrite right after here. Also, generator is always set in this case and not a flag, so there's no need to branch on whether it exists or not within the forward method.